packages/ra-core/src/controller/list/useInfiniteListController.ts

```diff
@@ -139,6 +143,33 @@ export const useInfiniteListController = <RecordType extends RaRecord = any>(
         }
     );
 
+    const { mutate: onSelectAll } = useMutation({
+        mutationFn: () =>
+            dataProvider.getList(resource, {
+                pagination: {
+                    page: 1,
+                    perPage: selectAllLimit,
+                },
+                sort: { field: query.sort, order: query.order },
+                filter: { ...query.filter, ...filter },
+                meta,
+            }),
+        onSuccess: ({ data }) => {
+            const allIds = data?.map(({ id }) => id) || [];
+            selectionModifiers.select(allIds);
+            if (allIds.length === selectAllLimit) {
+                notify('ra.message.too_many_elements', {
+                    messageArgs: { max: selectAllLimit },
+                    type: 'warning',
+                });
+            }
+        },
+        onError: e => {
+            console.error('Mutation Error: ', e);
+            notify('An error occurred. Please try again.');
+        },
+    });
```

Comment:
This should probably be a hook that could be called from both `useInfiniteListController` and `useListController`